rocketpy/control/controller.py

```diff
@@ -2,42 +2,58 @@
         be observed.
 
         Parameters
         ----------
-        observed_objects : list
-            A list of objects to be observed by the controller. It can be any
-            python object. This list will be passed to the controller function
-            as positional arguments, meaning that the order of the objects in
-            this list matters. These objects will then be able to be accessed
-            and modified by the controller function.
+        interactable_objects : list
+            A collection of objects that the controller function can access and
+            potentially modify. This list is flexible and can include any Python
+            object. The objects listed here are provided to the controller
+            function as the last argument, maintaining the order specified in
+            this list. The controller function gains the ability to interact
+            with and make adjustments to these objects during its execution.
         controller_function : function, callable
-            A function that takes the following arguments, in this order:
-
-            1. Time of the simulation at the current step in seconds.
-            2. The sampling rate at which the controller function is called in
-               Hertz (Hz).
-            3. The state vector of the simulation, which is defined as:
+            An user-defined function responsible for controlling the simulation.
+            This function is expected to take the following arguments, in order:
 
+            1. `time` (float): The current simulation time in seconds.
+            2. `sampling_rate` (float): The rate at which the controller
+               function is called, measured in Hertz (Hz).
+            3. `state` (list): The state vector of the simulation, structured as
                `[x, y, z, vx, vy, vz, e0, e1, e2, e3, wx, wy, wz]`.
-            4. A list containing the state history of the simulation. The state
-               history is a list of every state vector of every step of the
-               simulation. The state history is a list of lists, where each
-               sublist is a state vector and is ordered from oldest to newest.
-            5. A list containing the objects to be acted upon by the controller.
-               The objects in this list are the same as the objects in the
-               observed_objects list, but they can be modified by the controller.
+            4. `state_history` (list): A record of the rocket's state at each
+               step throughout the simulation. The state_history is organized as
+               a list of lists, with each sublist containing a state vector. The
+               last item in the list always corresponds to the previous state
+               vector, providing a chronological sequence of the rocket's
+               evolving states.
+            5. `observed_variables` (list): A list containing the variables that
```

Comment:
@MateusStano, about your comment in the description that the `observed_variables` may substitute the need of the `state`.

- If it were to substitute, every time a variable from the rocket state is needed the rocket would have to be added to the interactable objects;
- Due to the rocket state importance and ubiquity for control computation, I believe it is fair that it is a default argument of the controller function (`state`) for convenience.

You are indeed correct that the observable variables can substitute the `state`, but I do not think it is worth it. Let me know what you think.

Reply:
Okay, lets keep it then